engine/robot.py

```diff
@@ -468,38 +516,58 @@ def execute_boundary_following(self, min_dist):
                     * With a margin of error accounting for not uniform obstacles
         2. Once both sensors are displaying the same values, move forward (boundary following)
         3. Exits boundary following when conditions met in execute_avoid_obstacle() method
+
+        For now, we assume the side ultrasonic sensor values are from the actual sensors, so changes needed for is_sim
         '''
-        front_dist = self.robot_state.rf_ultrasonic.distance()
-        back_dist = self.robot_state.rb_ultrasonic.distance()
-        margin = 1 
-        # Plus or minus distance value used to calculate if robot is parallel to an non-uniform object 
-        # (i.e. not a flat surface). forwardRightSensorReading - backRightSensorReading < margin means
-        # robot is parallel to object.
-        forward_dist = 0.01  # Distance moved forward by robot in one iteration of this method
-        turn_angle = math.pi / 90
-        # Turn angle of robot in one iteration of this method. Robot turns turn_angle if
-        # forwardRightSensorReading > backRightSensorReading and -turn_angle if
-        # forwardRightSensorReading < backRightSensorReading
-
-        if math.abs(front_dist - back_dist) < margin:
-            self.move_forward(forward_dist)
-        elif front_dist > back_dist:
-            self.turn(turn_angle)
+        from electrical.ultrasonic_sensor import Ultrasonic
+        front_ultrasonic = Ultrasonic(0)
+        rf_ultrasonic = Ultrasonic(1)
+        rb_ultrasonic = Ultrasonic(2)
+        lf_ultrasonic = Ultrasonic(3)
+        # length of the robot in meters, placeholder
+        length = 10
+        # robot should be side_margin_sensor meters away from side obstacle, placeholder
+        side_margin_sensor = 1
```

Comment:
make this a global variable/ parameter?

Reply:
is this in reference to length? I will make it an attribute of robot_state